web-client/src/presenter/actions/AdvancedSearch/updateAdvancedSearchFormAction.ts

```diff
@@ -5,12 +5,29 @@ export const updateAdvancedSearchFormAction =
   ({
     props,
     store,
+    get,
   }: ActionProps<{
     formType: string;
     key: string;
     value: string | object;
+    isArray?: boolean;
   }>) => {
     const formType = formName || props.formType;
+
+    if (props.isArray) {
+      const currentValue =
+        get(state.advancedSearchForm[formType][props.key]) || [];
+      const updatedValue = currentValue.includes(props.value)
+        ? currentValue.filter((value: string) => value !== props.value)
+        : [...currentValue, props.value];
```

Comment:
I know that the trial sessions filters call a separate sequence entirely (`setTrialSessionsFiltersSequence`). I wonder if that is worth doing? 

It might make things more readable: we would know what state is in this specific form, we would avoid a boolean flag, and we could encapsulate certain assumptions we make in the form. Here, for instance, the assumption is that we want to toggle, i.e., if we re-add the same value, we should actually remove it. One might just as easily expect the value to be included just once, e.g., `[...new Set(blah blah)]`. This is a nitpick; I am just thinking out loud.